src/GeositeFramework/plugins/layer_selector_v2/WmsService.js

```diff
@@ -0,0 +1,139 @@
+define([
+        "dojo/_base/declare",
+        "dojo/Deferred",
+        "underscore",
+        "./lib/xmlToJson",
+        "framework/util/ajax",
+        "./util",
+        "./LayerNode",
+    ],
+    function(declare,
+             Deferred,
+             _,
+             xmlToJson,
+             ajaxUtil,
+             util,
+             LayerNode) {
+        "use strict";
+
+        return declare(null, {
+            constructor: function(server) {
+                this.server = server;
+            },
+
+            getServiceUrl: function() {
+                return util.urljoin(this._getBaseServiceUrl() + '?request=GetCapabilities&service=WMS');
+            },
+
+            _getBaseServiceUrl: function() {
+                // Likely an AGS WMS service
```

Comment:
AGS can publish services as REST services or WMS services. Our module `AgsService` is really `AgsRESTService`. If the WMS service is published on AGS, you need to specify the name of the service. There is no such thing (as far as I can tell) for Geoserver WMS services. I say likely because I'm not totally sure this will always hold up. It seems to be true for the services I tested.